markdown.dtx

```diff
@@ -24182,6 +24390,574 @@ function M.reader.new(writer, options)
 % \par
 % \begin{markdown}
 %
+%#### Parsers Used for Indentation (local)
```

Comment:
These long lists of helper functions should be definitely mentioned in the limitations section of the thesis. Practically speaking, the helper functions make the definition of syntax extensions much more difficult. That is because instead of having the logic available as individual PEG pattern building blocks, it is hidden in opaque functions. Pandoc already implements the whole of commonmark in PEG, so a reader familiar with Haskell should be able to find ways to refactor these helper functions into a more idiomatic PEG grammar.